<launch>
	<node name="detect_grasps_active" pkg="agile_grasp2" type="detect_grasps" output="screen">
    <!-- Input Point Cloud -->
    <param name="cloud_type" value="1" /> <!-- 0: *.pcd file, 1: sensor_msgs/PointCloud2, 2: agile_grasp/CloudSized, 3: agile_grasp/CloudIndexed -->
    <param name="use_service" value="false" />
    <param name="cloud_topic" value="/cloud_infinitam" />
        
    <!-- Plotting -->
    <param name="plot_mode" value="0" /> <!-- 0: none, 1: PCL, 2: rviz -->
    <param name="only_plot_output" value="false" />
        
    <!-- Hand Search -->
    <rosparam param="workspace"> [0.52, 0.92, -0.34, 0.05, -0.25, 1] </rosparam>
    <rosparam param="camera_pose"> [] </rosparam>
    <param name="num_samples" value="10000" />
    <param name="num_threads" value="4" />    
    <param name="nn_radius_taubin" value="0.01" />
    <param name="nn_radius_hands" value="0.1" />
    <param name="num_orientations" value="8" />
    <param name="antipodal_mode" value="1" /> <!-- 0: use hypotheses, 1: predict antipodal graps, 2: calculate antipodal grasps -->
    <param name="normal_estimation_method" value="1" /> <!-- 0: quadrics, 1: pcl::NormalEstimationOMP -->
    <param name="voxelize" value="false"/>
    <param name="filter_half_grasps" value="false"/>
    <param name="gripper_width_range" value="[0.03, 0.08]" />
    
    <!-- Hand Geometry -->
    <param name="finger_width" value="0.01" />
    <param name="hand_outer_diameter" value="0.09" />
    <param name="hand_depth" value="0.06" />
    <param name="hand_height" value="0.02" />
    <param name="init_bite" value="0.01" />
    
    <!-- Classifier -->
    <param name="images_directory" value="/home/andreas/data/grasp_images/" />
    <param name="model_file" value="$(find agile_grasp2)/caffe/test_1batch2.prototxt" />
    <param name="trained_file" value="$(find agile_grasp2)/caffe/bottles_boxes_cans_5xNeg.caffemodel" />
    <param name="label_file" value="$(find agile_grasp2)/caffe/labels.txt" />
    <!-- <param name="min_score_diff" value="650" /> March experiments 650 -->
    <param name="min_score_diff" value="300" />
    
    <!-- Grasp Selection -->
    <param name="num_selected" value="1000" />    
	</node>
</launch>
